anti_patterns/deny-warnings.md

```diff
@@ -0,0 +1,35 @@
+# `#![deny(warnings)]`
+
+## Description
+
+A well-intentioned crate author wants to ensure their code builds without
+warnings. So they annotate their crate root with
+
+## Example
+
+```rust
+#![deny(warnings)]
+
+// All is well.
+```
+
+## Drawbacks
+
+Until, say, the language changes, and as part of the transition, a lint is
+introduced that `warn`s, and Boom! The code no longer builds. Or someone tries
+to compile the code with additional lints (e.g. [clippy]), only to find out
+that they have to remove the annotation first.
+
+## Alternatives
+
+Our plucky developer hero should have used
+`RUSTFLAGS="-D warnings" cargo build"` to build their crate, removing
+the need for the annotation in their code.
+
+## See also
+
+- Type `rustc -W help` for a list of lints on your system. Also type
+`rustc --help` for a general list of options
+- [clippy] is a collection of lints for better Rust code
```

Comment:
I wonder if it is worth being explicit and calling it 'rust-clippy' rather than just clippy, since the latter is technically something else (and probably TM etc. MS).